scapy/contrib/isotp.py

```diff
@@ -718,46 +804,52 @@ def recv(self, x=0xffff):
 
     @staticmethod
     def select(sockets, remain=None):
+        # type: (List[SuperSocket], Optional[float]) -> Tuple[List[SuperSocket], None]  # noqa: E501
         """This function is called during sendrecv() routine to wait for
         sockets to be ready to receive
         """
         blocking = remain is None or remain > 0
 
-        def find_ready_sockets():
-            return list(filter(lambda x: (x.ins is not None) and
-                                         (x.ins.rx_queue is not None) and
-                                         (not x.ins.rx_queue.empty()),
-                               sockets))
+        def find_ready_sockets(socks):
+            # type: (List[ISOTPSoftSocket]) -> List[ISOTPSoftSocket]
+            return [x for x in socks if
+                    x.iso_ins is not None and
+                    x.iso_ins.rx_queue is not None and
+                    not x.iso_ins.rx_queue.empty()]
 
-        ready_sockets = find_ready_sockets()
+        iso_socks = [s for s in sockets if isinstance(s, ISOTPSoftSocket)]
```

Comment:
- I'm surprised you didn't need to type this.
- You could use a generator here

Reply:
Done

Reply:
I tried with a generator, but I use it again, later in the code.